sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -545,6 +574,187 @@ def set_read_only(
+        :type name: str
+        :keyword match_condition: The match condition to use upon the etag.
+        :type match_condition: ~azure.core.MatchConditions
+        :keyword str etag: Check if the Snapshot is changed. Set None to skip checking etag.
+        :return: The Snapshot returned from the service.
+        :rtype: ~azure.appconfiguration.Snapshot
+        """
+        try:
+            generated_snapshot = self._impl.update_snapshot(
+                name=name,
+                entity=SnapshotUpdateParameters(status=SnapshotStatus.ARCHIVED),
+                if_match=prep_if_match(etag, match_condition),
+                if_none_match=prep_if_none_match(etag, match_condition),
+                **kwargs
+            )
+            return Snapshot._from_generated(generated_snapshot)
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def recover_snapshot(
+        self,
+        name: str,
+        *,
+        match_condition: MatchConditions = MatchConditions.Unconditionally,
+        etag: Optional[str] = None,
+        **kwargs
+    ) -> Snapshot:
+        """Recover a configuration setting snapshot. It will update the status of a snapshot from "archived" to "ready".
+
+        :param name: The name of the configuration setting snapshot to recover.
+        :type name: str
+        :keyword match_condition: The match condition to use upon the etag.
+        :type match_condition: ~azure.core.MatchConditions
+        :keyword str etag: Check if the Snapshot is changed. Set None to skip checking etag.
+        :return: The Snapshot returned from the service.
+        :rtype: ~azure.appconfiguration.Snapshot
+        """
+        try:
+            generated_snapshot = self._impl.update_snapshot(
+                name=name,
+                entity=SnapshotUpdateParameters(status=SnapshotStatus.READY),
+                if_match=prep_if_match(etag, match_condition),
+                if_none_match=prep_if_none_match(etag, match_condition),
+                **kwargs
+            )
+            return Snapshot._from_generated(generated_snapshot)
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def get_snapshot(self, name: str, *, fields: Optional[List[str]] = None, **kwargs) -> Snapshot:
+        """Get a configuration setting snapshot.
+
+        :param name: The name of the configuration setting snapshot to retrieve.
+        :type name: str
+        :keyword list[str] fields: Specify which fields to include in the results. Leave None to include all fields.
+        :return: The Snapshot returned from the service.
+        :rtype: ~azure.appconfiguration.Snapshot
+        """
+        try:
+            generated_snapshot = self._impl.get_snapshot(
+                name=name, if_match=None, if_none_match=None, select=fields, **kwargs
```

Comment:
Why aren't we exposing the match_condition + etag here?

Reply:
Because we want to get the result unconditionally.